cpp/test/sync_comms_test.cpp

```diff
@@ -226,5 +226,41 @@ TEST_CASE("bcast table", "[sync comms]") {
   }
 }
 
+TEMPLATE_LIST_TEST_CASE("allreduce array", "[sync comms]", ArrowNumericTypes) {
+  auto type = default_type_instance<TestType>();
+  auto rank = *arrow::MakeScalar(RANK)->CastTo(type);
+  auto base_arr =  ArrayFromJSON(type, "[1, 2, 3, 4]");
+
+  // [1, 2, 3, 4] * rank
+  auto arr = arrow::compute::Multiply(base_arr, rank)->make_array();
+  auto col = Column::Make(std::move(arr));
+
+  const auto &comm = ctx->GetCommunicator();
+
+  auto test_allreduce = [&](net::ReduceOp op, const auto &exp) {
+    std::shared_ptr<Column> res;
+    CHECK_CYLON_STATUS(comm->AllReduce(col, op, &res));
```

Comment:
Nice. I am thinking, can we expose this API to Python and let people use MPI4Py with Cylon data structures. 
This would be a cool feature.

Reply:
Yeah, I think it would be a trivial change, now that all the `Column` and `Scalar` objects are there in cython

Reply:
@vibhatha can you open a JIRA for this?